src/Our.Umbraco.DocTypeGridEditor/Web/Controllers/DocTypeGridEditorApiController.cs

```diff
@@ -83,5 +84,35 @@ public object GetDataTypePreValues(string dtdId)
             var propEditor = PropertyEditorResolver.Current.GetByAlias(dtd.PropertyEditorAlias);
             return propEditor.PreValueEditor.ConvertDbToEditor(propEditor.DefaultPreValues, preValue);
         }
+
+        [System.Web.Http.HttpGet]
+        public List<PropertyType> GetContentTypePropertyTypes(String alias)
+        {
+            var list = new List<PropertyType>();
+            var doctype = UmbracoContext.Application.Services.ContentTypeService.GetContentType(alias);
+            if (doctype != null)
+            {
+                list = doctype.PropertyTypes.ToList();
+                if (doctype.ParentId != 0)
```

Comment:
You are right, something smells bad here.
Looks like if we need the sort order of the inherited properties we should be checking the CompositionPropertyGroups instead of the ParentId.

I will investigate and get back to you.